src/moapplication.cpp

```diff
@@ -81,12 +142,304 @@ MOApplication::MOApplication(int& argc, char** argv)
+    createVersionInfo().displayString(3), GITID,
+    QCoreApplication::applicationDirPath(), MOShared::getUsvfsVersionString());
+
+  if (singleInstance.secondary()) {
+    log::debug("another instance of MO is running but --multiple was given");
+  }
+
+  log::info("data path: {}", currentInstance->directory());
+  log::info("working directory: {}", QDir::currentPath());
+
+
+  // deleting old files, only for the main instance
+  if (!singleInstance.secondary()) {
+    purgeOldFiles();
+  }
+
+  QWindowsWindowFunctions::setWindowActivationBehavior(
+    QWindowsWindowFunctions::AlwaysActivateWindow);
+
+
+  // loading settings
+  Settings settings(currentInstance->iniPath(), true);
+  log::getDefault().setLevel(settings.diagnostics().logLevel());
+  log::debug("using ini at '{}'", settings.filename());
+
+  OrganizerCore::setGlobalCoreDumpType(settings.diagnostics().coreDumpType());
+
+
+  // logging and checking
+  env::Environment env;
+  env.dump(settings);
+  settings.dump();
+  sanity::checkEnvironment(env);
+
+  const auto moduleNotification = env.onModuleLoaded(qApp, [](auto&& m) {
+    log::debug("loaded module {}", m.toString());
+    sanity::checkIncompatibleModule(m);
+  });
+
+
+  // this must outlive `organizer`
+  std::unique_ptr<PluginContainer> pluginContainer;
+
+  // nexus interface
+  log::debug("initializing nexus interface");
+  NexusInterface ni(&settings);
+
+  // organizer core
+  log::debug("initializing core");
+  OrganizerCore organizer(settings);
+  if (!organizer.bootstrap()) {
+    reportError("failed to set up data paths");
+    InstanceManager::singleton().clearCurrentInstance();
+    return 1;
+  }
+
+  // plugins
+  log::debug("initializing plugins");
+  pluginContainer = std::make_unique<PluginContainer>(&organizer);
+  pluginContainer->loadPlugins();
+
+  // instance
+  if (auto r=setupInstanceLoop(*currentInstance, *pluginContainer)) {
+    return *r;
+  }
+
+  if (currentInstance->isPortable()) {
+    log::debug("this is a portable instance");
+  }
+
+  sanity::checkPaths(*currentInstance->gamePlugin(), settings);
+
+  // setting up organizer core
+  organizer.setManagedGame(currentInstance->gamePlugin());
+  organizer.createDefaultProfile();
+
+  log::info(
+    "using game plugin '{}' ('{}', variant {}, steam id '{}') at {}",
+    currentInstance->gamePlugin()->gameName(),
+    currentInstance->gamePlugin()->gameShortName(),
+    (settings.game().edition().value_or("").isEmpty() ?
+      "(none)" : *settings.game().edition()),
+    currentInstance->gamePlugin()->steamAPPId(),
+    currentInstance->gamePlugin()->gameDirectory().absolutePath());
+
+  CategoryFactory::instance().loadCategories();
+  organizer.updateExecutablesList();
+  organizer.updateModInfoFromDisc();
+  organizer.setCurrentProfile(currentInstance->profileName());
+
+  // checking command line
+  if (auto r=m_cl.setupCore(organizer)) {
+    return *r;
+  }
+
+  // show splash
+  MOSplash splash(

```

Comment:
Could the splash be showed earlier? I don't know if this can actually make a perceivable difference, I guess only if stuff before was slow enough. Just my instinct was to show it as soon as possible to give sign of life.

Reply:
This is the earliest point where 1) the instance is valid and a game plugin has been found, and 2) we know the command line isn't requesting to run something. We don't show splash when running a shortcut. I guess we could rewrite `setupCore()` to first figure out if something needs to run and show the splash before `updateExecutablesList()`, etc. if it doesn't, but I'm not sure it's worth it.

Reply:
At least, not in this PR.

Reply:
Yeah I agree.